templates/vmalert/deployment.yaml

```diff
@@ -1,77 +1,80 @@
+  {{- include "vm-benchmark.labels" $ | nindent 4 }}
 spec:
   selector:
     matchLabels:
-  {{- include "vm-benchmark.selectorLabels" . | nindent 6 }}
+  {{- include "vm-benchmark.selectorLabels" $ | nindent 6 }}
   template:
     metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/vmalert/vmalert-cm.yaml") $ | sha256sum }}
       labels:
-    {{- include "vm-benchmark.labels" . | nindent 8 }}
+   {{- include "vm-benchmark.selectorLabels" $ | nindent 8 }}
     spec:
       containers:
-        - name: {{ template "vm-benchmark.name" . }}-{{ .Values.vmalert.name }}
-          image: "{{ .Values.vmalert.image.repository }}:{{ .Values.vmalert.image.tag }}"
-          imagePullPolicy: "{{ .Values.vmalert.image.pullPolicy }}"
+        - name: vmalert
+          image: "{{ $.Values.vmalert.image.repository }}:{{ $.Values.vmalert.image.tag }}"
+          imagePullPolicy: "{{ $.Values.vmalert.image.pullPolicy }}"
           args:
-            - --httpListenAddr=:{{ .Values.vmalert.service.hostPort }}
-            - --notifier.url=http://127.0.0.1:{{ .Values.vmalert.service.amPort }}
+            - --httpListenAddr=:{{ $.Values.vmalert.port }}
+            - --notifier.url=http://127.0.0.1:{{ $.Values.vmalert.amPort }}
             - --rule=/config/alerts.yml
-            - --evaluationInterval={{ .Values.vmalert.evaluationInterval }}
-            - --datasource.url={{ .Values.vmalert.datasource.url }}
-            {{- if .Values.vmalert.datasource.token }}
-            - --datasource.bearerToken={{ .Values.vmalert.datasource.token }}
+            - --evaluationInterval={{ $.Values.vmalert.evaluationInterval }}
+            - --datasource.url={{ $url.datasource.url }}
+            {{- if $url.datasource.token }}
+            - --datasource.bearerToken={{ $url.datasource.token }}
             {{- end }}
-            {{- if .Values.vmalert.remoteWrite.url }}
-            - --remoteWrite.url={{ .Values.vmalert.remoteWrite.url }}
+            {{- if $url.remoteWrite.url }}
+            - --remoteWrite.url={{ $url.remoteWrite.url }}
             {{- end }}
-            {{- if .Values.vmalert.remoteWrite.token }}
-            - --remoteWrite.bearerToken={{ .Values.vmalert.remoteWrite.token }}
+            {{- if $url.remoteWrite.token }}
+            - --remoteWrite.bearerToken={{ $url.remoteWrite.token }}
           {{- end }}
           ports:
             - name: metrics
-              containerPort: {{ .Values.vmalert.service.hostPort }}
-              hostPort: {{ .Values.vmalert.service.hostPort }}
+              containerPort: {{ $.Values.vmalert.port }}
           resources:
-{{ toYaml .Values.vmalert.resources | indent 12 }}
+{{ toYaml $.Values.vmalert.resources | indent 12 }}
           volumeMounts:
             - name: vmalert-cfg
               mountPath: /config
 
-        - name: {{ template "vm-benchmark.name" . }}-alertmanager
+        - name: alertmanager
```

Comment:
so it starts N alertmanagers as well, correct?

Reply:
yep, i think, that overhead is very low it's ok to launch multiple alertmanager.